src/library/dao/playlistdao.cpp

```diff
@@ -150,6 +150,26 @@ QList<TrackId> PlaylistDAO::getTrackIds(const int playlistId) const {
     return trackIds;
 }
 
+QList<TrackId> PlaylistDAO::getTrackIdsInPlaylistOrder(const int playlistId) const {
```

Comment:
This method is a copy of `getTrackIds()`, but with explicit ordering of the returned track ids.  I couldn't see any equivalent method that already does this, and didn't want to break existing behaviour.

I've kept the `DISTINCT` in here, but I'm not sure why it was in the original `getTrackIds()` method.  Happy to be enlightened by someone in the know!

Reply:
Not an author of the original code, but I feel like the original `getTrackIds` was just supposed to return a mathematical set of all the trackids in the playlist. Since a mathematical set has no ordering and no duplicates, the `DISTINCT` was required. In this case, its up to you whether you want the list deduplicated or not. Since you rely on the ordering, its not a set anymore and it doesn't need its values deduplicated unless you want it to IMO.

Reply:
That's a fair comment.

I had forgotten that I actually expose whether the Engine database supports multiple tracks in a single playlist as a DB feature.  I've pushed a new commit that changes the behaviour:

1. Now `PlaylistDAO::getTrackIdsInPlaylistOrder()` does not de-duplicate.
2. In the playlist export, the DB is interrogated to see if it supports duplicate tracks in a playlist or not, and either exports the tracks verbatim or de-duplicates as needed.